assets/src/edit-story/components/inspector/documentInspector.js

```diff
@@ -1,9 +1,187 @@
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useEffect } from '@wordpress/element';
+
+/**
+ * External dependencies
+ */
+import styled, { css } from 'styled-components';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../app/story';
+import { useConfig } from '../../app/config';
+import UploadButton from '../uploadButton';
+import useInspector from './useInspector';
+import { SelectMenu, InputGroup } from './shared';
+
+const ButtonCSS = css`
+	color: ${ ( { theme } ) => theme.colors.mg.v1 };
+	font-size: 14px;
+	width: 100%;
+	padding: 15px;
+	background: none;
+	margin: 5px 0px;
+`;
+const Img = styled.img`
+	width: 100%;
+	max-height: 300px
+`;
+
+const Group = styled.div`
+	border-color: ${ ( { theme } ) => theme.colors.mg.v1 };
+	display: block;
+	align-items: center;
+	margin: 15px 0px;
+`;
+
+const RemoveButton = styled.button`
+	${ ButtonCSS }
+`;
 
 function DocumentInspector() {
+	const {
+		actions: { loadStatuses, loadUsers, setDisabledStatuses },
+		state: { users, statuses, disabledStatuses },
+	} = useInspector();
+
+	const {
+		state: { meta: { isSaving }, story: { author, status, slug, date, excerpt, featuredMediaUrl, password }, capabilities },
+		actions: { updateStory, deleteStory },
+	} = useStory();
+
+	const { postThumbnails } = useConfig();
+
+	useEffect( () => {
+		loadStatuses();
+		loadUsers();
+	} );
+
+	useEffect( () => {
+		const states = ( status === 'future' ) ? [ 'pending' ] : [ 'future', 'pending' ];
+		setDisabledStatuses( states );
+	}, [ status, setDisabledStatuses ] );
```

Comment:
Disabled statuses are only ever used in this component, are they not? Can this just be converted to:

```js
const disabledStatuses = useMemo( () => ( status === 'future' ) ? [ 'pending' ] : [ 'future', 'pending' ], [ status ] );
```

And then everything about disabled statues in the context and provider can be completely deleted? Unless it's required elsewhere of course.

Memoizing isn't even that important, but it looks nice and saves on some re-renders.

Reply:
Good shout. https://github.com/ampproject/amp-wp/pull/4122/commits/f6fd82f3eae64b9f1e3f01f29358ab967618091a